subsys/bluetooth/host/keys_br.c

```diff
@@ -52,31 +55,50 @@ struct bt_keys_link_key *bt_keys_get_link_key(const bt_addr_t *addr)
 	}
 
 	key = bt_keys_find_link_key(BT_ADDR_ANY);
-	if (key) {
-		bt_addr_copy(&key->addr, addr);
-		BT_DBG("created %p for %s", key, bt_addr_str(addr));
-
-		return key;
+	if (!key) {
+		if (IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)) {
+			bt_keys_link_key_clear(&key_pool[ARRAY_SIZE(key_pool) - 1]);
+			for (i = (ARRAY_SIZE(key_pool) - 1); i > 0; i--) {
+				memcpy(key_pool + i, key_pool + i - 1, sizeof(key_pool[0]));
+			}
+
+			key = &key_pool[0];
+		} else {
+			return NULL;
+		}
 	}
+	bt_addr_copy(&key->addr, addr);
+	BT_DBG("created %p for %s", key, bt_addr_str(addr));
 
-	BT_DBG("unable to create link key for %s", bt_addr_str(addr));
-
-	return NULL;
+	return key;
 }
 
 void bt_keys_link_key_clear(struct bt_keys_link_key *link_key)
 {
-	BT_DBG("%s", bt_addr_str(&link_key->addr));
+#if IS_ENABLED(CONFIG_BT_SETTINGS)
+	char key[BT_SETTINGS_KEY_MAX];
+	bt_addr_le_t le_addr;
+
+	le_addr.type = BT_ADDR_LE_PUBLIC;
+	bt_addr_copy(&le_addr.a, &link_key->addr);
+	bt_settings_encode_key(key, sizeof(key), "link_key", &le_addr, NULL);
```

Comment:
Since `bt_settings_encode_key` accepts `bt_addr_le_t` maybe it would be needed to have a `_br` version? Then again it is just one byte in addition